zerver/openapi/zulip.yaml

```diff
@@ -2558,34 +2612,39 @@ paths:
                       description: |
                         An object that contains `emoji` objects, each identified with their
                         emoji ID as the key.
-                      properties:
-                        id:
-                          type: integer
-                          description: |
-                            The ID for this emoji, same as the object's key.
-                        name:
-                          type: string
-                          description: |
-                            The user-friendly name for this emoji. Users in the organization
-                            can use this emoji by writing this name between colons (`:name:`).
-                        source_url:
-                          type: string
-                          description: |
-                            The path relative to the organization's URL where the
-                            emoji's image can be found.
-                        deactivated:
-                          type: string
-                          description: |
-                            Whether the emoji has been deactivated or not.
-                        author_id:
-                          type: integer
-                          nullable: true
-                          description: |
-                            The user ID of the user who uploaded the custom emoji.
-                            Will be null if the uploader is unknown.
-
-                            **Changes**: New in Zulip 2.2 (feature level 7).  Previously
-                            was accessible via and `author` object with an `id` field.
+                      additionalProperties:
```

Comment:
What's the story behind moving these from `properties` to `additionalProperties`?  I would have thought that these fields, which are all always present, should just stay with `properties`.  

(I guess I don't understand when one is supposed to use `additionalProperties` in general?)

Reply:
properties signifies that we know the quantity of that data. For example:
```
properties:
  obj:
    type: object
```
Will signify that there is only one key `obj` with a value of data type `object`. On the other hand if we don't know the name of the key or as in this case, we don't know the name of keys but they all share the same schema, then `additionalProperties` can be used.

Reply:
OK, maybe sense; maybe worth a comment above `additionalProperties` here explaining that?  Also, is there somewhere we communicate the type of the keys?

Reply:
Yes we have `description` attribute of `additionalProperties` for describing the keys. They are also in `get_presence`.

Reply:
In the api docs of [get_presence](https://zulipchat.com/api/get-presence) the defintion of `{client_name}` is made using `additionalProperties` 's description.

Reply:
Ahh, and I guess in JSON, the keys are always strings.